src/utils/wallet/index.ts

```
@@ -216,93 +215,91 @@ export const generateAddresses = async ({
-						type,
-					});
-					if (address.isErr()) {
-						return err(address.error.message);
-					}
-					const scriptHash = await getScriptHash(
-						address.value.address,
-						selectedNetwork,
-					);
-					if (!scriptHash) {
-						return err('Unable to get script hash.');
-					}
-					addresses[scriptHash] = {
-						...address.value,
-						index,
-						scriptHash,
-					};
-				} catch {}
-			}),
-		);
-		await Promise.all(
-			changeAddressArray.map(async (_item, i) => {
-				try {
-					const index = i + changeAddressIndex;
-					const path = { ...keyDerivationPath };
-					path.addressIndex = `${index}`;
-					const changeAddressPath = formatKeyDerivationPath({
-						path,
-						selectedNetwork,
-						accountType,
-						changeAddress: true,
-						addressIndex: `${index}`,
-					});
-					if (changeAddressPath.isErr()) {
-						return err(changeAddressPath.error.message);
-					}
 
-					const address = await getAddress({
-						path: changeAddressPath.value.pathString,
-						selectedNetwork,
-						type,
-					});
-					if (address.isErr()) {
-						return err(address.error.message);
-					}
-					const scriptHash = await getScriptHash(
-						address.value.address,
-						selectedNetwork,
-					);
-					if (!scriptHash) {
-						return err('Unable to get script hash.');
-					}
-					changeAddresses[scriptHash] = {
-						...address.value,
-						index,
-						scriptHash,
-					};
-				} catch {}
-			}),
-		);
+		// Generate Addresses
+		const addresses = {} as IAddresses;
+		const changeAddresses = {} as IAddresses;
+
+		for (let i = 0; i < addressAmount; i++) {
+			try {
+				const index = i + addressIndex;
+				let path = { ...keyDerivationPath };
+				path.addressIndex = `${index}`;
+				const addressPath = formatKeyDerivationPath({
+					path,
+					selectedNetwork,
+					accountType,
+					changeAddress: false,
+					addressIndex: `${index}`,
+				});
+				if (addressPath.isErr()) {
+					return err(addressPath.error.message);
+				}
```

If you return in map, you only skip one element
If you return in for loop, you exit from it
IMHO here should be return, but let's check if @coreyphillips

---

I think we should definitely let it error out and not ignore because it's critical that this does what it should. But need to handle errors here.